lib/Toolchain/Runner.cpp

```diff
@@ -82,3 +82,54 @@ ExecutionResult Runner::runProgram(const std::string &program,
 
   return result;
 }
+
+ExecutionResult Runner::runProgramFromWorkingDirectory(const std::string &program,
```

Comment:
I avoided extending the existing `runProgram` method with an optional `working directory` argument and simply created another focused method. Also, the existing method has code that does some manipulation with the environment, so I thought it made sense to remove it from the new `runProgramFromWorkingDirectory`. (Hope this is ok.)

Reply:
I have decided to merge them into one 😃